crates/engine/tree/src/tree/mod.rs

```diff
@@ -755,3 +846,36 @@ where
         todo!()
     }
 }
+
+/// The state of the persistence task.
+#[derive(Default, Debug)]
+struct PersistenceState {
+    /// Hash of the last block persisted. A None value means no persistence task
+    /// has run yet.
+    last_persisted_block_hash: Option<B256>,
+    /// Receiver end of channel where the result of the persistence task will be
+    /// sent when done. A None value means there's no persistence task in progress.
+    rx: Option<oneshot::Receiver<B256>>,
+    /// The last persisted block number.
+    last_persisted_block_number: u64,
+}
```

Comment:
I like this type - IMO given the hash is an `Option` it also makes sense for the last persisted block number to also be an `Option`

Reply:
good point, we should be consistent with the stored hash and block number. Given that we are not using the hash's Option for anything I have removed it, so both the hash and the block number are plain values now.

Reply:
makes sense!